<launch>
    <!-- USB config-->
    <arg default="/dev/ttyUSB0" name="uart_port"></arg>
    <arg default="/dev/ttyUSB1" name="data_port"></arg>

    <!-- Topic names-->
    <arg default="isk_1" name="radar_id"></arg>
    <arg default="/gtec/mmwave/$(arg radar_id)/radar" name="publish_radar_topic"></arg>
    <arg default="/gtec/mmwave/$(arg radar_id)/cloud" name="publish_cloud_topic"></arg>
    <arg default="/gtec/mmwave/$(arg radar_id)/target" name="publish_target_topic"></arg>
    <arg default="/gtec/mmwave/$(arg radar_id)/cartesian" name="publish_cartesian_cloud_topic"></arg>
    <arg default="/gtec/mmwave/all/target" name="publish_all_target_topic"></arg>
    <arg default="/gtec/mmwave/all/cartesian" name="publish_all_cartesian_topic"></arg>

    <!-- Radar device config-->

    <arg default="$(find gtec_mmwave_reader)/example_radar_configs/people_counting_14m_6843ISK.cfg" name="config_file_path"></arg>
    

    <!-- People Counting parameters (angles in degrees) -->
    <arg default="5" name="elev_tilt"></arg>

    <!-- Radar Pose within ROS (angles in radiands) -->
    <arg default="true" name="ignore_elevation"></arg>
    
    <arg default="12.46" name="radar_pos_x"></arg>
    <arg default="3.624" name="radar_pos_y"></arg>
    <arg default="2.335" name="radar_pos_z"></arg>
    <arg default="-3.1416" name="radar_yaw"></arg>
    <arg default="0.0873" name="radar_pitch"></arg>
    <arg default="0" name="radar_roll"></arg>

    <arg default="-3.624" name="boundary_box_x_min"></arg>
    <arg default="2.775" name="boundary_box_x_max"></arg>

    <arg default="0.5" name="boundary_box_y_min"></arg>
    <arg default="13.5" name="boundary_box_y_max"></arg>

    <arg default="0" name="boundary_box_z_min"></arg> 
    <arg default="2.4" name="boundary_box_z_max"></arg>

    <!-- Nodes -->

    <node  name="IWR6843ISKPeopleCountingReader_$(arg radar_id)" output="screen" pkg="gtec_mmwave_reader" type="IWR6843ISK_people_counting_reader.py">
        <param name="uart_port" value="$(arg uart_port)"></param>
        <param name="data_port" value="$(arg data_port)"></param>
        <param name="config_file_path" value="$(arg config_file_path)"></param>
        <param name="publish_radar_topic" value="$(arg publish_radar_topic)"></param>
        <param name="publish_cloud_topic" value="$(arg publish_cloud_topic)"></param>
        <param name="publish_target_topic" value="$(arg publish_target_topic)"></param>
        <param name="sensor_height" value="$(arg radar_pos_z)"></param>
        <param name="elev_tilt" value="$(arg elev_tilt)"></param>
        <param name="radar_yaw" value="$(arg radar_yaw)"></param>
        <param name="radar_pitch" value="$(arg radar_pitch)"></param>
        <param name="radar_roll" value="$(arg radar_roll)"></param>
        <param name="radar_id" value="$(arg radar_id)"></param>
        <param name="publish_all_target_topic" value="$(arg publish_all_target_topic)"></param>
        <param name="sensor_x" value="$(arg radar_pos_x)"></param>
        <param name="sensor_y" value="$(arg radar_pos_y)"></param>
        <param name="boundary_box_x_min" value="$(arg boundary_box_x_min)"></param>
        <param name="boundary_box_y_min" value="$(arg boundary_box_y_min)"></param>
        <param name="boundary_box_z_min" value="$(arg boundary_box_z_min)"></param>
        <param name="boundary_box_x_max" value="$(arg boundary_box_x_max)"></param>
        <param name="boundary_box_y_max" value="$(arg boundary_box_y_max)"></param>
        <param name="boundary_box_z_max" value="$(arg boundary_box_z_max)"></param>
    </node>

   <node pkg="tf" 
        type="static_transform_publisher" 
        name="odom_to_radar_$(arg radar_id)" 
        args="$(arg radar_pos_x) $(arg radar_pos_y) $(arg radar_pos_z) $(arg radar_yaw) $(arg radar_pitch) $(arg radar_roll) odom $(arg radar_id) 30" />

    <node  pkg="tf" type="static_transform_publisher" name="map_to_odom_$(arg radar_id)" args="0 0 0 0 0 0 map odom 30" />

<node  pkg="tf" type="static_transform_publisher" name="odom_to_target_$(arg radar_id)" args="$(arg radar_pos_x) $(arg radar_pos_y) 0 0 0 0 odom target_$(arg radar_id) 30" />

    <node  pkg="pose_publisher" type="pose_publisher" name="radar_pose_$(arg radar_id)">
            <param name="map_frame" value="odom"></param>
            <param name="base_frame" value="$(arg radar_id)"></param>
            <remap from="/pose" to="/gtec/mmwave/$(arg radar_id)/pose"/>
    </node>


    <node name="IWR6843ISKPolarToCartesian_$(arg radar_id)" output="screen" pkg="gtec_mmwave_reader" type="IWR6843ISK_polar_to_cartesian.py">
        <param name="radar_topic" value="$(arg publish_radar_topic)"></param>
        <param name="radar_id" value="$(arg radar_id)"></param>
        <param name="publish_cloud_topic" value="$(arg publish_cartesian_cloud_topic)"></param>
        <param name="publish_all_cartesian_topic" value="$(arg publish_all_cartesian_topic)"></param>
        <param name="radar_yaw" value="$(arg radar_yaw)"></param>
        <param name="radar_pitch" value="$(arg radar_pitch)"></param>
        <param name="radar_roll" value="$(arg radar_roll)"></param>
        <param name="ignore_elevation" value="$(arg ignore_elevation)"></param>
    </node>

</launch>